sparkly/writer.py

```diff
@@ -255,6 +275,188 @@ def write_partition_to_kafka(messages):
+                 'flush': Flush all existing data before writing.
+            host (str|None): Redis host. Either this or redis_client_init
+                must be provided. See below.
+            port (int|6379): Port redis is listening to.
+            db (int|0): Redis db to write to.
+            redis_client_init (callable|None): Bypass internal redis
+                 client initialization by passing a function that does it,
+                 no arguments required. For example this could be
+                 redis.StrictRedis.from_url with the appropriate url and
+                 kwargs already set through functools.partial. This option
+                 overrides other conflicting arguments.
+
+        Raises:
+            NotImplementedError, if redis-py is not installed.
+            AssertionError, if nor host neither redis_client_init are
+                provided.
+            ValueError, if any of the expire, compression, max pipeline
+                size or mode options assume an invalid value.
+        """
+        if not REDIS_WRITER_SUPPORT:
+            raise NotImplementedError(
+                'redis package is not available. Use pip install sparkly[redis] to fix it.'
+            )
+
+        assert host or redis_client_init, \
+            'redis: At least one of host or redis_client_init must be provided.'
+
+        if expire is not None and expire < 1:
+            raise ValueError('redis: expire must be positive')
+
+        if compression not in {None, 'bzip2', 'gzip', 'zlib'}:
+            raise ValueError(
+                'redis: bzip2, gzip and zlib are the only supported compression codecs.'
+            )
+
+        if max_pipeline_size < 1:
+            raise ValueError('redis: max pipeline size must be positive')
+
+        if mode not in {'overwrite', 'ignore', 'flush'}:
+            raise ValueError(
+                'redis: only overwrite (default), ignore and flush modes are supported.'
+            )
+
+        key_name = '_sparkly_redis_key_col_{}'.format(uuid.uuid4().hex)
+        value_name = '_sparkly_redis_value_col_{}'.format(uuid.uuid4().hex)
+
+        # Compute the key
+        df = self._df.withColumn(
+            key_name,
+            F.concat_ws(
+                key_delimiter,
+                *(
+                    ([F.lit(key_prefix).astype(T.StringType())] if key_prefix else []) +
+                    [F.col(col_name).astype(T.StringType()) for col_name in key_by]
+                )
+            )
+        )
+
+        if exclude_key_columns:
+            for col_name in key_by:
+                df = df.drop(col_name)
+
+        if group_by_key:
+            df = (
+                df
+                .withColumn(
+                    value_name,
+                    F.struct(*[col for col in df.columns if col != key_name])
+                )
+                .groupBy(key_name)
+                .agg(F.expr('collect_list( {} )'.format(value_name)).alias(value_name))
```

Comment:
Good catch, stayed as an `expr` because I was using brickhouse's `collect` for this in spark 1.6